hypothesis-python/src/hypothesis/strategies.py

```diff
@@ -926,7 +928,30 @@ def text(
     elif isinstance(alphabet, SearchStrategy):
         char_strategy = alphabet
     else:
-        char_strategy = sampled_from(list(map(text_type, alphabet)))
+        if not isinstance(alphabet, Sequence):
+            note_deprecation(
+                'alphabet must be an ordered sequence, or tests may be '
+                'flaky and shrinking weaker, but a %r is not a type of '
+                'sequence.  This will be an error in future.'
+                % (type(alphabet),)
+            )
+        alphabet = list(alphabet)
+        non_unicode = [c for c in alphabet if not isinstance(c, string_types)]
```

Comment:
This variable should now be called `non_string` rather than `non_unicode`.  However, the error message should retain it's reference to "unicode strings" to avoid people trying to use byte-strings on Python 3.

You will also need to add a test that using an alphabet with non-string contents is deprecated (e.g. `alphabet=[1, 2, 3]`) to achieve full test coverage.

Reply:
got it.